src/Nethermind/Nethermind.Optimism/OptimismHeaderValidator.cs

```diff
@@ -36,5 +37,25 @@ public class OptimismHeaderValidator(
         new PreBedrockHeaderValidator(blockTree, sealValidator, specProvider, logManager),
         blockTree, specProvider, sealValidator, logManager)
 {
+    public override bool Validate(BlockHeader header, BlockHeader? parent, bool isUncle, out string? error)
+    {
+        IReleaseSpec spec = _specProvider.GetSpec(header);
+        if (spec.IsOpHoloceneEnabled)
+        {
+            try
+            {
+                // TODO: We might want to avoid using exceptions here
```

Comment:
As noted, we might want to avoid exceptions. If we go that route, then we should change the API to `TryParseEIP1559Parameters(..., out EIP1559Parameters)`, the conventional C# pattern.

Reply:
Yeah pleas do avoid exceptions!

Reply:
Changed to errors over Exceptions.